contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,128 @@
+    /// (LastEpochStart)
+    function totalVotingStake() external view returns (uint256);
+
+    /// @notice Look up the balance of `blsVK`
+    function lookup(BN254.G1Point calldata blsVK) external view returns (uint64);
+    /// @notice Look up the full `Node` state associated with `blsVK`
+    function fullLookup(BN254.G1Point calldata blsVK) external view returns (Node memory);
+
+    // === Queuing Stats ===
+
+    /// @notice Get the next available epoch for new registration
+    function nextRegistrationEpoch() external view returns (uint64);
+    /// @notice Get the number of pending registration requests in the waiting queue
+    function numPendingRegistrations() external view returns (uint64);
+    /// @notice Get the next available epoch for exit
+    function nextExitEpoch() external view returns (uint64);
+    /// @notice Get the number of pending exit requests in the waiting queue
+    function numPendingExit() external view returns (uint64);
+
+    // === Write APIs ===
+
+    /// @notice Register a validator in the stake table, transfer of tokens incurred!
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param schnorrVK The Schnorr verification key (as the auxiliary info)
+    /// @param amount The amount to register
+    /// @param stakeType The type of staking (native or restaking)
+    /// @param maxWait The maximum epoch the sender is waiting to wait to be included (cannot be
+    /// smaller than the current epoch)
+    ///
+    /// @return success status
+    ///
+    /// @dev No validity check on `schnorrVK`, as it's assumed to be sender's responsibility,
+    /// the contract only treat it as auxiliary info submitted by `blsVK`.
+    function register(
+        BN254.G1Point calldata blsVK,
+        EdOnBN254.EdOnBN254Point calldata schnorrVK,
+        uint64 amount,
+        StakeType stakeType,
+        uint64 maxWait
+    ) external returns (bool);
+
+    /// @notice Deposit more stakes to registered keys
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param amount The amount to deposit
+    /// @return (newBalance, effectiveEpoch) the new balance effective at a future epoch
+    function deposit(BN254.G1Point calldata blsVK, uint64 amount)
+        external
+        returns (uint64, uint64);
+
+    /// @notice Request to exit from the stake table, not immediately withdrawable!
+    ///
+    /// @param blsVK The BLS verification key to exit
+    /// @return success status
+    function exit(BN254.G1Point calldata blsVK) external returns (bool);
+
+    /// @notice Withdraw from the staking pool. Transfers occur! Only successfully exited keys can
+    /// withdraw past their `exitEpoch`.
+    ///
+    /// @param blsVK The BLS verification key to withdraw
+    /// @return The total amount withdrawn, equal to `Node.balance` associated with `blsVK`
+    function withdraw(BN254.G1Point calldata blsVK) external returns (uint64);
```

Comment:
```suggestion
    function withdrawFunds(BN254.G1Point calldata blsVK) external returns (uint64);
```
Since `withdraw` can have multiple meanings.

Reply:
done in [8ff8e8d](https://github.com/EspressoSystems/espresso-sequencer/pull/748/commits/8ff8e8d77fb6eebd42fd313f36fab14add4b423d)